log4j-api/src/main/java/org/apache/logging/log4j/status/StatusConsoleListener.java

```diff
@@ -134,7 +165,7 @@ public Level getStatusLevel() {
     @Override
     public void log(final StatusData data) {
         requireNonNull(data, "data");
-        if (level.isLessSpecificThan(data.getLevel())) {
+        if (debugEnabled || level.isLessSpecificThan(data.getLevel())) {
```

Comment:
I think the correct fix is to remove the filtering on level here. `StatusLogger` is already taking care of the filtering, listeners should not (incorrectly!) duplicate that work. Hence, instead of leaking `debugEnabled` to `StatusConsoleListener`, could you remove this `if` block, please?

As a result of this change, `StatusConsoleListenerTest#level_and_stream_should_be_honored` test will be broken. Could you rename it to `stream_should_be_honored` and fix it too, please? There you won't need multiple messages with different levels anymore. Checking the dump of a single message should be good enough.

I can share some development tips to quickly iterate over your changes:
* Compile your `log4j-api` changes: `./mvnw clean install -Dmaven.test.skip -Dspotless.skip -Dcyclonedx.skip -Dspotbugs.skip -Drat.skip -Dbnd.baseline.skip -pl :log4j-api`
* Make sure you fixed `StatusConsoleListenerTest`: `./mvnw test -Dspotless.skip -Dcyclonedx.skip -Dspotbugs.skip -Drat.skip -Dbnd.baseline.skip -pl :log4j-api-test -Dtest=StatusConsoleListenerTest`
* Fix formatting: `./mvnw spotless:apply -pl :log4j-api,:log4j-api-test`
* Make sure all is in good shape: `./mvnw clean install -pl :log4j-api,:log4j-api-test`